src/test/java/FakeConsole.java

```diff
@@ -20,8 +21,12 @@ public FakeConsole(String nextLine, int nextInt) {
 
     @Override
     public int nextInt() {
-        if(nextInt==null){
-            return Integer.parseInt(nextLine);
+        try {
+            if(nextInt==null){
+                return Integer.parseInt(nextLine);
+            }
+        } catch (Exception exception) {
```

Comment:
you don't need this. It is implementation for test. It should throw exception ONLY when you want it in your test. And in such case you will define this exception explicitly. Like you do in shouldBeThrownException test. If you don't want exception in your test implementation, but it is happened, then you should fix it. And does not matter which type of exception is thrown.